### PR TITLE
Honor REPLACE_OS_VARS on upgrade

### LINKED ISSUE #139: REPLACE_OS_VARS doesn't work with upgrades
### Steps to reproduce

- Start a release with `REPLACE_OS_VARS=true`
- Check `Application.get_env` returns interpolated values
- Run an upgrade
- Check same `Applucation.get_env` call

### Verbose Logs

```
> MIX_ENV=prod mix release
> rel/three_six_five/bin/three_six_five start

Interactive Elixir (1.3.4) - press Ctrl+C to exit (type h() ENTER for help)
iex(three_six_five@127.0.0.1)1> Application.get_all_env(:jobs)
[{Jobs.Repo,
  [adapter: Ecto.Adapters.Postgres, database: "jobs", hostname: "localhost",
   username: "brenton", password: ""]}, {:included_applications, []},
 {:ecto_repos, [Jobs.Repo]}]

> gc --allow-empty -m "touch"
> MIX_ENV=prod mix release --upgrade
> REPLACE_OS_VARS=true rel/three_six_five/bin/three_six_five upgrade 614d89a

iex(three_six_five@127.0.0.1)2> Application.get_all_env(:jobs)
[{Jobs.Repo,
  [adapter: Ecto.Adapters.Postgres, database: "jobs",
   hostname: "${DB_HOSTNAME}", username: "${DB_USERNAME}",
   password: "${DB_PASSWORD}"]}, {:included_applications, []},
 {:ecto_repos, [Jobs.Repo]}]
```

### Description of issue

- What are the expected results?

ENV vars are interpolated, even if `REPLACE_OS_VARS` required for upgrade

- What OS, Erlang/Elixir versions are you seeing this issue on?

Arch Linux
Erlang/OTP 19 [erts-8.1.1] [source] [64-bit] [smp:4:4] [async-threads:10] [hipe] [kernel-poll:false]
IEx 1.3.4

- If possible, also provide your `rel/config.exs`, as it is often
  my first troubleshooting question, and you'll save us both time :)

```
use Mix.Releases.Config,
    default_release: :my_app,
    default_environment: :prod

environment :prod do
  set include_erts: true
  set include_src: false
  set cookie: System.get_env("BEAM_COOKIE")
  set pre_start_hook: "rel/hooks/pre_start"
  set post_start_hook: "rel/hooks/post_start"
end

release :my_app do
  set version: "#{System.cmd("git", ["rev-parse", "--short", "HEAD"]) |> elem(0) |> String.rstrip}"
  set applications: [
    rollbax: :permanent,
    jobs: :permanent,
  ]
end
```

- Is there documentation that says one thing, but Distillery does
  another? If so, please link the doc here so it can be updated if
  it's a documentation issue, or so that the fix can be based around
  what's documented.

I can't find any documentation regarding `REPLACE_OS_VARS` and upgrades, unfortunately.

### PR DESCRIPTION
**Pull request by shosti**
The Erlang hot upgrade mechanism seems to only look for sys.config in
the release directory, so the pre-processed sys.config was getting
ignored. This change should still allow custom release configs to
work (which wasn't the case previously IIUC), as well as replacing OS
vars as expected.

Please let me know if this looks like the right approach (I'm pretty new
to Erlang release management in general).

Fixes #139

### CODE REVIEW

**Comment by shosti** [comment]
Is there anything else that needs to be done for this, out of curiosity?

**Comment by mkaszubowski** [comment]
Just bumping this because it would be really helpful to have this available :) Is there anything that needs to be done before merging?

**Comment by lorenzo** [comment]
I can report that this patch correctly fixes the issue :)

**Comment by bitwalker** [comment]
Sorry it took so long, I hate GitHub's review system sometimes, because "pending" comments don't show up for the submitter until I submit the review :/

File: priv/templates/boot.eex [hunk 1]
```diff
@@ -272,6 +272,9 @@ fi
 # By default it's RELEASE_ROOT_DIR
 if [ -z "$RELEASE_CONFIG_DIR" ]; then
     RELEASE_CONFIG_DIR="$RELEASE_ROOT_DIR"
+    CUSTOM_CONFIG_DIR=""
+else
+    CUSTOM_CONFIG_DIR="true"
```
Comment (shosti) [inline priv/templates/boot.eex]: It's to track whether a custom config dir is in use (i.e. `RELEASE_CONFIG_DIR` was set before the script got to this point). Will change to `HAS_CUSTOM_CONFIG_DIR`.

**Comment by bitwalker** [comment]
@shosti ping

**Comment by bitwalker** [comment]
I've handled this work manually. Please give version 1.2.2 a shot everyone!

**Pull request closed without merging**